pkg/kubelet/pod_manager.go

```diff
@@ -114,6 +115,17 @@ func (pm *basicPodManager) UpdatePods(u PodUpdate, podSyncTypes map[types.UID]me
 		}
 		allPods := applyUpdates(u.Pods, pm.getAllPods())
 		pm.setPods(allPods)
+
+	case cache.ADD:
+		glog.V(3).Infof("Add: Containers changed")
+		for i := range u.Pods {
+			podSyncTypes[u.Pods[i].UID] = metrics.SyncPodCreate
+		}
+		// TODO: Something about thread safety needs thinking about here.
```

Comment:
Hmm, no i meant, we were previously deep copying the new pods in the merge function (I'm skipping the deep copy now since it can be cpu intensive) and i need to stare at it some more to see if there was an actual reason why